scripts/githooks/check-file-format.sh

```diff
@@ -48,18 +48,11 @@ function main() {
   else
 
     # Check changed files only
-    files=$( (git diff --diff-filter=ACMRT --name-only ${BRANCH_NAME:-origin/main}; git diff --name-only) | sort | uniq )
-    if [ -n "$files" ]; then
-      while read file; do
-        docker run --rm --platform linux/amd64 \
-          --volume=$PWD:/check \
-          mstruebing/editorconfig-checker:$image_version \
-            ec \
-              --exclude '.git/' \
-              "$file"
-        [ $? != 0 ] && exit_code=1 ||:
-      done < <(echo "$files")
-    fi
+    docker run --rm --platform linux/amd64 \
+      --volume=$PWD:/check \
+      mstruebing/editorconfig-checker:$image_version \
+        sh -c 'ec --exclude ".git/" $(git diff --diff-filter=ACMRT --name-only)'
```

Comment:
Can we change this line to:

```suggestion
        sh -c "ec --exclude '.git/' \$(git diff --diff-filter=ACMRT --name-only ${BRANCH_NAME:-origin/main})"
```

The branch name is required when run as action.

Reply:
The branch name has a very good chance of being wrong when run locally.  I think it needs to be smarter than that but I'm drawing a blank as to how.

Reply:
I suspect we might actually want two different invocations here.  Locally you're much more likely to only want to check the thing you're currently trying to commit.  As a github action you probably want to check every patch since the merge-base.

Reply:
How do we feel about adding a `mode` parameter to `check-file-format.sh`?  Because we can move the file selection back out of the container with an invocation like this:

```shell
ec --exclude ".git/" $(cat /dev/stdin)
``` 

There are two things we need to make that work: `--interactive` on the `docker run` invocation so that `stdin` gets passed through, and a `scripts/config/editorconfig-checker.sh` wrapper for the `ec` command so the shell quoting doesn't get completely unmanageable.  But with those in place, we could have something like this in `check-file-format.sh`:

```shell
    file_list_command="git diff --diff-filter=ACMRT --name-only"
    if [[ $mode == "git-action" ]]; then file_list_command="${file_list_command} ${BRANCH_NAME:-origin/main}"; fi

    $file_list_command | docker run --rm -i --volume=${PWD}:/check $image /check/scripts/config/editorconfig-invocation.sh
```

Does that make sense?  Is it better than what we've got?  It solves the problem of the upstream branch name potentially being wrong locally, while specifying it where we control the context.  I think I'd prefer it if we could determine which mode we were in from the execution context but I'm aware that could be fragile.

Reply:
With regard to a potential workaround of setting the `BRANCH_NAME` variable manually in forked repositories, that doesn't work: it's not the `BRANCH_NAME` that's wrong.  It's the `origin`.  Besides which, even if it *did* work there would be no indication to the user that they needed to do it.  It would just be a silent degradation.  That doesn't really work.

Reply:
Sorry, just realised I've misread slightly - yes, setting `BRANCH_NAME` would work, functionally, on a forked repository.  I still think that's not a reasonable workaround, again because there's no way a user would know they're in a situation where they need to.

Reply:
Sorry, accidentally marked this one as resolved.  Still going!